### ISSUE TITLE
Why no external vector library?

### ISSUE THREAD

**Issue by itmuckel** [mrDIMAS/rg3d#38]
Hey, great project!

I was wondering why you implemented vectors etc. yourself and not use the nalgebra crate for example? Any features missing there?

More context maybe: I need to rotate a vector and hoped to do a matrix-vector multiplication, but as far as I see there's no way.

**Comment by mrDIMAS** [comment]
Hello! The answer is very simple - I started this project as a port of my engine written in C, so I just ported all the stuff and continued adding more features. So far I added all the stuff that I need.

>  I need to rotate a vector and hoped to do a matrix-vector multiplication, but as far as I see there's no way.
There is [Mat4::transform_vector](https://github.com/mrDIMAS/rg3d-core/blob/master/src/math/mat4.rs#L254) which doing exactly what you want. I agree that having a method instead of operator is confusing, but this is how it is implemented.
